classes/class-wc-connect-nux.php

```diff
@@ -83,5 +97,307 @@ public function register_add_service_to_zone_pointer( $pointers ) {
 			);
 			return $pointers;
 		}
+
+		public function get_jetpack_install_status() {
+			// check if Jetpack is activated
+			if ( ! class_exists( 'Jetpack_Data' ) ) {
+				// not activated, check if installed
+				if ( 0 === validate_plugin( 'jetpack/jetpack.php' ) ) {
+					return self::JETPACK_INSTALLED;
+				}
+				return self::JETPACK_UNINSTALLED;
+			} else if ( defined( 'JETPACK_DEV_DEBUG' ) && true === JETPACK_DEV_DEBUG ) {
+				// installed, activated, and dev mode on
+				return self::JETPACK_DEV;
+			}
+
+			// installed, activated, dev mode off
+			// check if connected
+			$user_token = Jetpack_Data::get_access_token( JETPACK_MASTER_USER );
+			if ( isset( $user_token->external_user_id ) ) { // always an int
+				return self::JETPACK_CONNECTED;
+			}
+
+			return self::JETPACK_ACTIVATED;
+		}
+
+		public function should_display_nux_notice_on_screen( $screen ) {
+			if ( // Display if on any of these admin pages.
+				( // Products list.
+					'product' === $screen->post_type
+					&& 'edit' === $screen->base
+				)
+				|| ( // Orders list.
+					'shop_order' === $screen->post_type
+					&& 'edit' === $screen->base
+					)
+				|| ( // Edit order page.
+					'shop_order' === $screen->post_type
+					&& 'post' === $screen->base
+					)
+				|| ( // WooCommerce settings.
+					'woocommerce_page_wc-settings' === $screen->base
+					)
+				|| ( // WooCommerce featured extension page
+					'woocommerce_page_wc-addons' === $screen->base
+					&& isset( $_GET['section'] ) && 'featured' === $_GET['section']
+					)
+				|| ( // WooCommerce shipping extension page
+					'woocommerce_page_wc-addons' === $screen->base
+					&& isset( $_GET['section'] ) && 'shipping_methods' === $_GET['section']
+					)
+				|| 'plugins' === $screen->base
+			) {
+				return true;
+			}
+			return false;
+		}
+
+		public function should_display_nux_notice_for_current_store_locale() {
+			$base_location = wc_get_base_location();
+			$country = isset( $base_location['country'] )
+				? $base_location['country']
+				: '';
+			// Do not display for non-US, non-CA stores.
+			if ( 'CA' === $country || 'US' === $country ) {
+				return true;
+			}
+			return false;
+		}
+
+		public function get_jetpack_redirect_url() {
+			$full_path = add_query_arg( array() );
+			// Remove [...]/wp-admin so we can use admin_url().
+			$new_index = strpos( $full_path, '/wp-admin' ) + strlen( '/wp-admin' );
+			$path = substr( $full_path, $new_index );
+			return admin_url( $path );
+		}
```

Comment:
That doesn't work for sites installed in a subfolder, it will output the subfolder twice in the URL:

`http://example.com/wp/wp/wp-admin/plugins.php`

This is because `add_query_arg( array() )` returns the queries _and_ the full path. So, if the WordPress install is in a subfolder, the result will look like this: `/wp/wp-admin/plugins.php`. 

`site_url()` uses the URL for the current site where WordPress files are installed, and appends the path passed in. Since they both contain the subfolder, the result has the subfolder twice.

I looked through the code for [`wp_safe_redirect`](https://core.trac.wordpress.org/browser/tags/4.8/src/wp-includes/pluggable.php#L1281) and related functions, but WordPress does not get the full URL to do the redirects, only the path passed in, since relative URLs will work.

We can't pass in just the path (i.e. the result of `add_query_arg( array() )`) here because Jetpack expects a full URL for the redirect. Otherwise, it will end up at `wordpress.com/$path_passed_in` after the connection is done.